src/wallet/scriptpubkeyman.cpp

```diff
@@ -732,12 +732,28 @@ bool LegacyScriptPubKeyMan::AddWatchOnly(const CScript& dest, int64_t nCreateTim
 void LegacyScriptPubKeyMan::SetHDChain(const CHDChain& chain, bool memonly)
 {
     LOCK(cs_wallet);
-    if (!memonly && !WalletBatch(m_storage.GetDatabase()).WriteHDChain(chain))
-        throw std::runtime_error(std::string(__func__) + ": writing chain failed");
+    if (!memonly) {
+        if (!WalletBatch(m_storage.GetDatabase()).WriteHDChain(chain)) {
+            throw std::runtime_error(std::string(__func__) + ": writing chain failed");
+        }
+        if (!hdChain.seed_id.IsNull()) {
+            AddInactiveHDChain(hdChain, false);
```

Comment:
Pass `memonly`? (even if value is always `false` for now)

Reply:
Done